### PR TITLE
Fixes to location display

### PR DESCRIPTION
**Pull request by theresenieva**
Make improvements to how location information is displayed on the map.

Previously, the page had to be refreshed to see the new vote count or new location. I've made changes so that the vote count and location are displayed immediately on the map.

### CODE REVIEW

File: GoMeet/src/main/webapp/js/meeting-location-dao.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ class MeetingLocationDAO {
     params.append('lat', lat);
     params.append('lng', lng);
     params.append('note', note);
-    let result = await fetch(this.storingEndPoint, {method: 'POST', body: params})
-      .then((response) => response.json());
+    let response = await fetch(this.storingEndPoint, {method: 'POST', body: params});
+    let result = await response.json();
```
Comment (vivianho18) [inline GoMeet/src/main/webapp/js/meeting-location-dao.js:44]: Why did you change from arrow function?

**Review by vivianho18 — COMMENTED** [review]
looks good :))

File: GoMeet/src/main/webapp/js/meeting-location-dao.js (hunk 1 above)
Comment (theresenieva) [reply GoMeet/src/main/webapp/js/meeting-location-dao.js:44]: I received a comment on a previous PR that using two await statements is cleaner. It is just a style choice.

Comment (vivianho18) [reply GoMeet/src/main/webapp/js/meeting-location-dao.js:44]: Oh my bad! - I thought we were told to preferentially use arrow functions for these async calls

Comment (theresenieva) [reply GoMeet/src/main/webapp/js/meeting-location-dao.js:44]: Haha nw!

File: GoMeet/src/main/webapp/js/interactive-map.js [hunk 2]
```diff
@@ -110,24 +109,31 @@ async function fetchLocations(map) {
...
-  noteContainer.innerText = note;
+  const titleContainer = createSpanContainer(title, 'displayTitle');
+  const noteContainer = createSpanContainer(note, 'displayNote');
+  const voteContainer = createSpanContainer(voteCount, 'displayVoteCount');
 
   const button = document.createElement('button');
+  button.setAttribute('id', 'voteButton');
   button.appendChild(document.createTextNode('VOTE'));
   button.onclick = () => {
     MeetingLocationDAO.updateLocation(keyString);
```
Comment (anna-giang) [inline GoMeet/src/main/webapp/js/interactive-map.js]: Did you need an `async` keyword here?

File: GoMeet/src/main/webapp/js/interactive-map.js [hunk 3]
```diff
@@ -110,24 +109,31 @@ async function fetchLocations(map) {
...
+    const currVote = voteContainer.innerText;
+    voteContainer.innerText = parseInt(currVote) + 1;
```
Comment (anna-giang) [inline GoMeet/src/main/webapp/js/interactive-map.js:122]: The code is directly updating the HTML display voteCount, which could be dangerous in that if the `MeetingLocationDAO.updateLocation()` call fails to update (error occurs during update), then the voteCount in the UI is inconsistent with the voteCount in the server. 

Will this be refactored later on to re-render the page to update the voteCount instead of directly updating it?

Comment (theresenieva) [reply GoMeet/src/main/webapp/js/interactive-map.js:122]: Yep, that's a good point. I do plan to refactor but instead I plan to make updateLocation() check the status of the response. If it is not 200 then updateLocation() will throw an error. In the button.onclick() it would look like this:

```
try {
    await MeetingLocationDAO.updateLocation(keyString);
    const currVote = voteContainer.innerText;
    voteContainer.innerText = parseInt(currVote) + 1;
} catch (error) {
  handleError();
}
```

That way the innerText is only updated if the updateLocation() does not throw an error. 

Does this seem like a good idea?

File: GoMeet/src/main/webapp/js/interactive-map.js (hunk 2 above)
Comment (theresenieva) [reply GoMeet/src/main/webapp/js/interactive-map.js]: Ahhh, yup. I do. Thanks for pointing that out :)

File: GoMeet/src/main/webapp/js/interactive-map.js (hunk 3 above)
Comment (anna-giang) [reply GoMeet/src/main/webapp/js/interactive-map.js:122]: Yep sounds good! I think its okay for this PR, but optionally, if you want to be extra safe, you could just not update the votes here, and have the user refresh the page to see the updated votes.

**Review by anna-giang — APPROVED** [review]
LGTM 👍

File: GoMeet/src/main/webapp/test/spec/spec-interactive-map.js [hunk 4]
```diff
@@ -120,7 +138,7 @@ describe ('Build Info Window Vote', function() {
   const TITLE_A = 'Taco Place';
   const COUNT_A = 2;
   const NOTE_A = 'Tacos taste yum!';
-
+
```
Comment (thanhdng) [inline GoMeet/src/main/webapp/test/spec/spec-interactive-map.js]: nit: please remove the spaces here

**Review by thanhdng — APPROVED** [review]
LGTM w nit
